hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/TrashPolicyOzone.java

```diff
@@ -0,0 +1,292 @@
+            " Using an interval of " +
+            (deletionInterval / MSECS_PER_MINUTE) +
+            " minutes that is used for deletion instead");
+        this.emptierInterval = deletionInterval;
+      }
+      LOG.info("Ozone Manager trash configuration: Deletion interval = "
+          + (deletionInterval / MSECS_PER_MINUTE)
+          + " minutes, Emptier interval = "
+          + (this.emptierInterval / MSECS_PER_MINUTE) + " minutes.");
+      executor = new ThreadPoolExecutor(TRASH_EMPTIER_CORE_POOL_SIZE,
+          TRASH_EMPTIER_CORE_POOL_SIZE, 1,
+          TimeUnit.SECONDS, new ArrayBlockingQueue<>(1024),
+          new ThreadPoolExecutor.CallerRunsPolicy());
+    }
+
+    @Override
+    public void run() {
+      if (emptierInterval == 0) {
+        return;                                   // trash disabled
+      }
+      long now, end;
+      while (true) {
+        now = Time.now();
+        end = ceiling(now, emptierInterval);
+        try {
+          // sleep for interval
+          Thread.sleep(end - now);
+        } catch (InterruptedException e) {
+          break;                                  // exit on interrupt
+        }
+
+        try {
+          now = Time.now();
+          if (now >= end) {
+            Collection<FileStatus> trashRoots;
+            trashRoots = fs.getTrashRoots(true); // list all trash dirs
+            LOG.info("TrashrootSize: " + trashRoots.size());
+            for (FileStatus trashRoot : trashRoots) {  // dump each trash
+              LOG.info("Trashroot:" + trashRoot.getPath().toString());
+              if (!trashRoot.isDirectory()) {
+                continue;
+              }
+              Runnable task = ()->{
+                try {
+                  TrashPolicyOzone trash = new TrashPolicyOzone(fs, conf);
+                  trash.deleteCheckpoint(trashRoot.getPath(), false);
+                  trash.createCheckpoint(trashRoot.getPath(),
+                          new Date(Time.now()));
+                } catch (IOException e) {
+                  LOG.info("Unable to checkpoint");
+                }
+              };
+              executor.submit(task);
+              LOG.info("Current threads in pool: "
+                  + executor.getPoolSize());
+              LOG.info("Currently executing threads: "
+                  + executor.getActiveCount());
+              LOG.info("Total number of threads(ever scheduled): "
+                  + executor.getTaskCount());
+            }
+          }
+        } catch (Exception e) {
+          LOG.warn("RuntimeException during Trash.Emptier.run(): ", e);
+        }
+      }
+      try {
+        fs.close();
+      } catch(IOException e) {
+        LOG.warn("Trash cannot close FileSystem: ", e);
+      }
+    }
+
+
+    private long ceiling(long time, long interval) {
+      return floor(time, interval) + interval;
+    }
+    private long floor(long time, long interval) {
+      return (time / interval) * interval;
+    }
+
```

Comment:
Can you add a shutdown method for executor thread pool? And we can shutdown executor thread firstly, then interrupted the emptier thread. This will be a better way to smoothly shutdown emptier thread